packages/moonstone/VirtualList/VirtualListBase.js

```diff
@@ -434,32 +451,62 @@ const VirtualListBaseFactory = (type) => {
-
-					if (nextIndex >= 0) {
-						ev.preventDefault();
-						ev.stopPropagation();
-						this.onAcceleratedKeyDown({isWrapped, keyCode, nextIndex, repeat, target});
+					const {repeat} = ev;
+					const {dimensionToExtent, isPrimaryDirectionVertical} = this.uiRefCurrent;
+					const isScrollButton = target.classList.contains(scrollbarCss.scrollButton);
+					const targetIndex = target.dataset.index;
+					const index = !isScrollButton ? getNumberValue(targetIndex) : -1;
+					const {isDownKey, isUpKey, isLeftMovement, isRightMovement, isWrapped, nextIndex} = this.getNextIndex({index, keyCode, repeat});
+					const directions = {};
+
+					if (isPrimaryDirectionVertical) {
+						directions.left = isLeftMovement;
+						directions.right = isRightMovement;
+						directions.up = isUpKey;
+						directions.down = isDownKey;
 					} else {
-						const {dataSize} = this.props;
-						const {dimensionToExtent} = this.uiRefCurrent;
-						const column = index % dimensionToExtent;
-						const row = (index - column) % dataSize / dimensionToExtent;
-						const isLeaving = isBackward && row === 0 ||
-							isForward && row === Math.floor((dataSize - 1) % dataSize / dimensionToExtent) ||
-							isLeftMovement && column === 0 ||
-							isRightMovement && column === dimensionToExtent - 1;
-
-						if (repeat && isLeaving) {
-							ev.preventDefault();
-							ev.stopPropagation();
-						} else if (!isLeaving && Spotlight.move(direction)) {
+						directions.left = isUpKey;
+						directions.right = isDownKey;
+						directions.up = isLeftMovement;
+						directions.down = isRightMovement;
+					}
+
+					if (!isScrollButton) {
+						if (nextIndex >= 0) {
 							ev.preventDefault();
 							ev.stopPropagation();
-							this.onAcceleratedKeyDown({keyCode, nextIndex: getNumberValue(Spotlight.getCurrent().dataset.index), repeat, target});
+							this.onAcceleratedKeyDown({isWrapped, keyCode, nextIndex, repeat, target});
+						} else {
+							const {dataSize, focusableScrollbar} = this.props;
+							const column = index % dimensionToExtent;
+							const row = (index - column) % dataSize / dimensionToExtent;
+							const isLeaving = directions.up && row === 0 ||
+								directions.down && row === Math.floor((dataSize - 1) % dataSize / dimensionToExtent) ||
+								directions.left && column === 0 ||
+								directions.right && !focusableScrollbar && column === dimensionToExtent - 1;
+
+							if (repeat && isLeaving) {
+								ev.preventDefault();
+								ev.stopPropagation();
+							} else if (!isLeaving && Spotlight.move(direction)) {
+								const nextTargetIndex = Spotlight.getCurrent().dataset.index;
+
+								ev.preventDefault();
+								ev.stopPropagation();
+
+								if (typeof nextTargetIndex === 'string') {
+									this.onAcceleratedKeyDown({keyCode, nextIndex: getNumberValue(nextTargetIndex), repeat, target});
+								}
+							}
+
 						}
+					} else if (
+						directions.right && repeat ||
```

Comment:
Don't we need to test `Spotlight.move` for `directions.right`? For example, for scenes like two lists in a single screen in channel edit?

Reply:
I don't believe so. When we reach this point in the code (the `else if` in the line above), focus is on a scroll button. The conditionals here determine only when we should prevent event bubbling. In this line `directions.right && repeat`, we want to prevent bubbling on repeated events, since focus should stop at the border during a key-hold. If we release and press 5way right (remembering that "right" means "down" in a horizontal list) again then none of these conditions would be truthy, so we wouldn't prevent event bubbling. At that point, spotlight would process the key event and move focus in the direction pressed.

Let me know if this makes sense or if I'm overlooking an edge-case.

Reply:
👍 It makes sense.